core/src/ast/types.rs

```diff
@@ -357,24 +364,35 @@ impl From<&syn::Type> for TypeName {
     /// - If the type is a path equal to [`diplomat_runtime::DiplomatResult`], returns a [`TypeName::Result`] with the type parameters recursively converted
     /// - If the type is a path equal to [`diplomat_runtime::DiplomatWriteable`], returns a [`TypeName::Writeable`]
     /// - If the type is a reference to `str`, returns a [`TypeName::StrReference`]
+    /// - If the type is a reference to a slice of a Rust primitive, returns a [`TypeName::PrimitiveSlice`]
     /// - If the type is a reference (`&` or `&mut`), returns a [`TypeName::Reference`] with the referenced type recursively converted
     /// - Otherwise, assume that the reference is to a [`CustomType`] in either the current module or another one, returns a [`TypeName::Named`]
     fn from(ty: &syn::Type) -> TypeName {
         match ty {
             syn::Type::Reference(r) => {
                 if r.elem.to_token_stream().to_string() == "str" {
-                    TypeName::StrReference
-                } else {
-                    TypeName::Reference(Box::new(r.elem.as_ref().into()), r.mutability.is_some())
+                    return TypeName::StrReference;
+                }
+                if let syn::Type::Slice(slice) = &*r.elem {
```

Comment:
I changed it to `return` because I need a double-layered match with dereferences, and I don't know how to do that with a single match:

```rust
if let syn::Type::Slice(slice) = &*r.elem {
    if let syn::Type::Path(p) = &*slice.elem {
        // do stuff
    }
}
```

Both the outer and the inner `if` should fall through to the outer `else`.